### PR TITLE
Refactor the model tests code in onnxruntime_test_all.exe

### PR DESCRIPTION
**Pull request by snnn**
**Description**: 

Update the code to use OrtApis instead of the old onnxruntime::InferenceSession class. Mainly because the old one doesn't support custom op.  We are trying to convert some EPs to custom ops. Hopefully they can continue to leverage this test set. 

**Motivation and Context**
- Why is this change required? What problem does it solve?
- If it fixes an open issue, please link to the issue here.

### CODE REVIEW

File: onnxruntime/test/providers/cpu/model_tests.cc [hunk 1]
```diff
@@ -25,6 +52,12 @@
 
 extern std::unique_ptr<Ort::Env> ort_env;
 
+#define ASSERT_ORT_STATUS_OK(function)   \
+  do {                                   \
+    OrtStatus* _tmp_status = (function); \
+    ASSERT_EQ(_tmp_status, nullptr);     \
```
Comment (edgchen1) [inline onnxruntime/test/providers/cpu/model_tests.cc:58]: is it possible to include the status code and message in the assertion failure output?

Comment (edgchen1) [inline onnxruntime/test/providers/cpu/model_tests.cc:58]: is it possible to include the status code and message in the assertion failure output?

Comment (snnn) [reply onnxruntime/test/providers/cpu/model_tests.cc:58]: > should a non-nullptr _tmp_status be freed

It doesn't matter. This is a unit test. When an assert fails, the developer should stop there and debug it. We don't need to care if memory leak would happen after an assert failed.

Comment (snnn) [reply onnxruntime/test/providers/cpu/model_tests.cc]: Error message added.

File: onnxruntime/test/providers/cpu/model_tests.cc [hunk 2]
```diff
@@ -714,6 +804,12 @@ TEST_P(ModelTest, Run) {
             break;
           }
         }
+        for (char* p : output_names) {
```
Comment (edgchen1) [inline onnxruntime/test/providers/cpu/model_tests.cc:807]: nit: could this clean up of output_names and output_values be done via RAII too?

**Review by edgchen1 — APPROVED** [review]

File: onnxruntime/test/providers/cpu/model_tests.cc (hunk 2 above)
Comment (snnn) [reply onnxruntime/test/providers/cpu/model_tests.cc:807]: Thanks. I will do it later.

**Pull request merged**